src/runtime_src/core/common/query_requests.h

```diff
@@ -660,6 +663,35 @@ struct kds_cu_info : request
   get(const device*) const = 0;
 };
 
+struct kds_mode : request
+{
+  using result_type = uint32_t;
+  static const key_type key = key_type::kds_mode;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct kds_cu_stat : request
+{
+  // Returning CUs statistic as <index, name, base_addr, status, usages>
+  using result_type = std::vector<std::tuple<uint32_t, std::string, uint64_t, uint32_t, uint64_t>>;
```

Comment:
Is this too implicit?  Can you replace with an embedded struct that leaves the comment unnecessary?

Reply:
This two lines are same as existed code in "struct kds_cu_info". So I thought this would be okay.
I can add embedded struct to avoid comment here.

Reply:
Got it yes.  I think we should change this.  It's a little magical, isn't it?

Reply:
fixed